src/mercury_engine_data_structures/version_validation.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from hashlib import md5
+from typing import TYPE_CHECKING
+
+from mercury_engine_data_structures.formats.toc import Toc
+from mercury_engine_data_structures.game_check import GameVersion
+
+if TYPE_CHECKING:
+    from mercury_engine_data_structures.file_tree_editor import FileTreeEditor
+
+
+def _get_md5(data: bytes) -> bytes:
+    return md5(data, usedforsecurity=False).digest()
+
+
+def identify_version(editor: FileTreeEditor) -> GameVersion:
```

Comment:
This function should receive the RomFS directly, to not receive a FileTreeEditor in mid-constructor (a thing you should avoid!)